corehq/apps/userreports/models.py

```diff
@@ -755,12 +758,35 @@ def rebuild_failed(self):
     def rebuild_awaiting_or_in_progress(self):
         return self.meta.build.awaiting or (self.meta.build.is_rebuild_in_progress and not self.rebuild_failed)
 
+    def set_rebuild_flags(self):
+        from .rebuild import get_table_diffs
+
+        if not self.rebuild_awaiting_or_in_progress:
```

Comment:
The logic after this looks solid.
I am a bit unsure about this check here.

If there is a build awaiting, we don't need to set flags again, makes sense.
if there is a build in progress though, should not we set `awaiting` again?
Additionally, `rebuild_failed` is quite unreliable. I will be adding ticket to remove its usage soon.

Reply:
I see this was updated later to rely only on awaiting, so this is good.